### PR TITLE
[CDAP-15787] Add failure collector for Kafka streaming source

### PR DESCRIPTION
**Pull request by CuriousVini**

### CODE REVIEW

File: kafka-plugins-0.10/pom.xml [hunk 1]
```diff
@@ -19,6 +19,16 @@
       <artifactId>kafka-plugins-common</artifactId>
       <version>${project.parent.version}</version>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.plugin</groupId>
+      <artifactId>format-common</artifactId>
```
Comment (albertshau) [inline kafka-plugins-0.10/pom.xml]: why does it need format-common?

File: kafka-plugins-0.10/pom.xml [hunk 2]
```diff
@@ -19,6 +19,16 @@
...
+      <version>${cdap.plugin.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>io.cdap.plugin</groupId>
+      <artifactId>format-avro</artifactId>
```
Comment (albertshau) [inline kafka-plugins-0.10/pom.xml]: why does it need format-avro? If avro is missing we should add it directly and not indirectly through this.

File: kafka-plugins-0.10/src/main/java/io/cdap/plugin/source/KafkaStreaminSourceUtil.java [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Util method for {@link KafkaStreamingSource}.
+ *
+ * This class contains methods for {@link KafkaStreamingSource} that require spark classes because during validation
+ * spark classes are not available. Refer CDAP-15912 for more information.
+ */
+final class KafkaStreaminSourceUtil {
```
Comment (albertshau) [inline kafka-plugins-0.10/src/main/java/io/cdap/plugin/source/KafkaStreaminSourceUtil.java]: typo: Streamin -> Streaming

**Review by albertshau — COMMENTED** [review]
didn't take a close look at the utils, assuming they are straight copies.

**Comment by CuriousVini** [comment]
yes utils are copies to avoid loading spark classes during validation except where ever methods with failure collector is being called, we are passing failure collector and throwing the exceptions.

File: kafka-plugins-0.10/pom.xml (hunk 2 above)
Comment (CuriousVini) [reply kafka-plugins-0.10/pom.xml]: fixed

File: kafka-plugins-0.10/pom.xml (hunk 1 above)
Comment (CuriousVini) [reply kafka-plugins-0.10/pom.xml]: fixed

**Review by albertshau — APPROVED** [review]

**Pull request merged**